pkg/module_manager/module_manager.go

```diff
@@ -1045,6 +1058,24 @@ func (mm *moduleManager) UpdateModuleDynamicValuesPatches(moduleName string, val
 		valuesPatch)
 }
 
+func (mm *moduleManager) ApplyModuleDynamicValuesPatches(moduleName string, values utils.Values) (utils.Values, error) {
+	mm.valuesLayersLock.Lock()
+	defer mm.valuesLayersLock.Unlock()
```

Comment:
do you really need write lock here?
Seems like you are reading patches map here and RLock is enough here

Reply:
Changed to RLock.